src/libraries/System.Reflection.DispatchProxy/src/System/Reflection/DispatchProxyGenerator.cs

```diff
@@ -394,9 +394,25 @@ internal void AddInterfaceImpl([DynamicallyAccessedMembers(DynamicallyAccessedMe
             private MethodBuilder AddMethodImpl(MethodInfo mi, int methodInfoIndex)
             {
                 ParameterInfo[] parameters = mi.GetParameters();
-                Type[] paramTypes = ParamTypes(parameters, false);
+                Type[] paramTypes = new Type[parameters.Length];
+                Type[][] paramReqMods = new Type[paramTypes.Length][];
+                Type[][] paramOptMods = new Type[paramTypes.Length][];
+
+                for (int i = 0; i < parameters.Length; i++)
+                {
+                    paramTypes[i] = parameters[i].ParameterType;
+                    paramReqMods[i] = parameters[i].GetRequiredCustomModifiers();
+                    paramOptMods[i] = parameters[i].GetOptionalCustomModifiers();
+                }
+
+                ParameterInfo returnParameter = mi.ReturnParameter;
+                Type[] returnReqMods = returnParameter.GetRequiredCustomModifiers();
+                Type[] returnOptMods = returnParameter.GetOptionalCustomModifiers();
+
+                MethodBuilder mdb = _tb.DefineMethod(mi.Name, MethodAttributes.Public | MethodAttributes.Virtual, CallingConventions.HasThis,
+                    returnParameter.ParameterType, returnReqMods, returnOptMods,
+                    paramTypes, paramReqMods, paramOptMods);
```

Comment:
Thanks, @wzchua , looks good to me.
To avoid side effects I would pass null for unnecessary parameters, plus i believe it's better to use the CallingConventions.Standard which was used by default in the previous call
```suggestion
                MethodBuilder mdb = _tb.DefineMethod(mi.Name, MethodAttributes.Public | MethodAttributes.Virtual, CallingConventions.Standard,
                    mi.ReturnType, null, null,
                    paramTypes, paramReqMods, null);
```

Reply:
ok. I thought it would be more correct for all the mods to be included.
I made a mistake with the CallingConventions.

Reply:
> ok. I thought it would be more correct for all the mods to be included.

Understand your motive, but adding those could affect performance, as we don't have a known issue with not adding them I prefer keep the existing behavior